frontend/src/components/VersionSelector/VersionSelector.tsx

```diff
@@ -0,0 +1,117 @@
+import { Button, Outline, P } from "@maykin-ui/admin-ui";
+import { FC, ReactNode, useState } from "react";
+
+import "./VersionSelector.css";
+
+type Version = {
+  id: string;
+  label: string;
+  isCurrent?: boolean;
+  isConcept?: boolean;
+};
+
+type VersionContent = {
+  id: string;
+  children: ReactNode;
+};
+
+interface VersionSelectorProps {
+  versions: Version[];
+  content: VersionContent[];
+}
+
+export const VersionSelector: FC<VersionSelectorProps> = ({
+  versions,
+  content,
+}) => {
+  const currentVersion = versions.find((v) => v.isCurrent);
+  const conceptVersion = versions.find((v) => v.isConcept);
+
+  const [isExpanded, setIsExpanded] = useState(false);
+  const [selectedVersionId, setSelectedVersionId] = useState(
+    currentVersion?.id || versions[0]?.id,
+  );
+
+  const selectedContent = content.find((item) => item.id === selectedVersionId);
+
+  const getVisibleVersions = () => {
+    // If expanded, show all versions
+    // If not expanded, show only the current and concept versions, and also the selected version
+    if (isExpanded) {
+      return versions;
+    }
+    return versions.filter(
+      (v) =>
+        v.isCurrent ||
+        v.isConcept ||
+        v.id === selectedVersionId ||
+        v.id === currentVersion?.id ||
+        v.id === conceptVersion?.id,
+    );
+  };
+
+  return (
+    <section
+      className="version-selector"
+      aria-labelledby="version-selector-heading"
+    >
+      <h2
+        id="version-selector-heading"
```

Comment:
If we hardcode the id' we will not be able to properly re-use this component. Please generated the (base) id using `useId()` so that we get unique id's.